src/main/java/picard/sam/markduplicates/util/OpticalDuplicateFinder.java

```diff
@@ -164,33 +170,32 @@ public OpticalDuplicateFinder(final String readNameRegex, final int opticalDupli
                 // and this loop seemed to kill the JVM for some reason. This logging statement (and the one in the
                 // loop below) solved the problem.
                 if (logProgress) progressLoggerForKeeper.record(String.format("%d", other.getReadGroup()), other.getX());
+                if (closeEnough(thisLoc, other, distance)) {
+                    opticalDistanceRelationGraph.addEdge(i, j);
+                }
             }
         }
```

Comment:
You might want to add debug level logging statements for building the graph and finishing building the graph.